R/annotate.R

```diff
@@ -0,0 +1,131 @@
+#' Annotates messy dates for uncertainty and/or approximation
+#'
+#' Some datasets have for example an arbitrary cut off point
+#' for start and end points, but these are often coded as precise dates
+#' when they are not necessarily the real start or end dates.
+#' This collection of functions helps annotate uncertainty and
+#' approximation to dates according to ISO2019E standards.
+#' Inaccurate start or end dates can be represented by an affix
+#' indicating "on or before", if used as a prefix (e.g. ..1816-01-01),
+#' or indicating "on or after", if used as a suffix (e.g. 2016-12-31..).
+#' Approximate dates are indicated by adding a `~` to year,
+#' month, or day components, as well as groups of components or whole dates
+#' to estimate values that are possibly correct (e.g. 2003-03-03~).
+#' Day, month, or year, uncertainty can be indicated by adding a `?`
+#' to a possibly dubious date (e.g. 1916-10-10?) or date
+#' component (e.g. 1916-?10-10).
+#' @param x A date vector
+#' @param component Annotation can be added on specific date components
+#' ("year", "month" or "day"), or to groups of date components (month and
+#' day ("md"), or year and month ("ym")). This must be specified.
+#' If unspecified, annotation will be added after the date (e.g. 1916-10-10?),
+#' indicating the whole date is uncertain or approximate.
+#' For specific date components, uncertainty or approximation is annotated to
+#' the left of the date component.
+#' E.g. for "day": 1916-10-?10 or 1916-10-~10.
+#' For groups of date components, uncertainty or approximation is annotated to
+#' the right of the group ("ym") or to both components ("md").
+#' E.g. for "ym": 1916-10~-10; for "md": 1916-?10-?10.
```

Comment:
For year-month, it is either added both before year and month components, or after the month component. I used the after month option because where possible, group annotation (i.e. to the right of the component) is preferred over individual annotation (to the left of the component) in the ISO guidelines. I used the latter option for month-date because adding the annotation to the right of date would make it the same as annotating the whole date as uncertain/approximate. Do you think it is better to make it consistent by adding it to the left for both year-month and month-date?